tests/test_random_number_generator.h

```diff
@@ -106,6 +105,81 @@ TEST_CASE("[RandomNumberGenerator] Restore from seed") {
 	CHECK_MESSAGE(s0_1_before == s0_1_after, msg);
 	CHECK_MESSAGE(s0_2_before == s0_2_after, msg);
 }
+
+TEST_CASE("[RandomNumberGenerator] Check rand range for expected functionality") {
+	Ref<RandomNumberGenerator> rng = memnew(RandomNumberGenerator);
+	rng->set_seed(0);
+	uint64_t prev_state = rng->get_state();
+	uint32_t seed_before = rng->get_seed();
+
+	String msg = "Randf range should give float between -100 and 100, base test";
+	float num_before = rng->randf_range(-100, 100);
+	CHECK_MESSAGE(-100 <= num_before, msg);
+	CHECK_MESSAGE(num_before <= 100, msg);
+	float num_after1 = rng->randf_range(-75, 75);
+	rng->randomize();
+
+	String msg2 = "Randf range should give float between -75 and 75.";
+	msg2 += " Shouldn't be affected by randomize.";
+	CHECK_MESSAGE(-75 <= num_after1, msg2);
+	CHECK_MESSAGE(num_after1 <= 75, msg2);
+	rng->set_state(prev_state);
+	int num_after2 = rng->randi_range(-50, 50);
```

Comment:
For this test suite to achieve good reliability, it's better that assertions should be done in a for loop with enough iterations due to pseudo-random nature of RNG, something like:

```c++
for (int i = 0; i < 1000; ++i) {
    int n = rng->randi_range(-50, 50);
    CHECK(n >= -50);
	CHECK(n <= 50);
}
```

Otherwise there's a high chance that the test case will miss a bug in case of regression.